### PR TITLE
Add support for Gradle 5

### PR DESCRIPTION
**Pull request by henriquenfaria**

### CODE REVIEW

**Comment by henriquenfaria** [comment]
@vanniktech Only tested this on a project running Gradle 5.1.1. Tomorrow I'll test this with older projects running Gradle 4.x.

File: src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy [hunk 1]
```diff
@@ -236,16 +236,14 @@ class GenerationPlugin implements Plugin<Project> {
 
             if (combined) {
                 // add instrumentation coverage execution data
-                executionData += subProject.fileTree("${subProject.buildDir}/outputs/code_coverage").matching {
```
Comment (vanniktech) [inline src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy]: Why was this removed?

File: src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy [hunk 2]
```diff
@@ -236,16 +236,14 @@ class GenerationPlugin implements Plugin<Project> {
...
-                    include "**/*.ec"
-                }
+
             }
 
             // add if true in extension or for the unit test Jacoco task
             def addToMergeTask = !combined || extension.includeInstrumentationCoverageInMergedReport
 
             if (mergeTask != null && addToMergeTask) {
-                mergeTask.executionData += executionData
```
Comment (vanniktech) [inline src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy]: What's the reason behind changing from adding to replacing?

**Comment by vanniktech** [comment]
Thanks for taking a stab on this!

File: src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy (hunk 1 above)
Comment (henriquenfaria) [reply src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy]: my bad!

File: src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy (hunk 2 above)
Comment (henriquenfaria) [reply src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy]: Another mistake of mine. I've refactored the whole PR to fix that. The StackOverFlow root cause is the += operator as explained [here](https://github.com/gradle/gradle/issues/8398).

**Comment by henriquenfaria** [comment]
Thanks for reviewing it @vanniktech ! Interesting to see the tests passing even though the implementation was wrong... Anyway, I've refactored it. Please, check again :)

File: src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy [hunk 3]
```diff
@@ -159,7 +159,7 @@ class GenerationPlugin implements Plugin<Project> {
                                       JacocoMerge mergeTask, JacocoReport mergedReportTask, final String taskName,
                                       final String jvmTestTaskName, final String instrumentationTestTaskName, final String sourceName,
                                       final String sourcePath, final String productFlavorName, final String buildTypeName) {
-        final def destinationDir
+        def destinationDir
```
Comment (vanniktech) [inline src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy]: Why was the `final` changed?

Comment (henriquenfaria) [reply src/main/groovy/com/vanniktech/android/junit/jacoco/GenerationPlugin.groovy]: Because this is not a valid assignment starting from Gradle 5.
If I leave the `final` keyword there I get the following compile error: `164: The variable [destinationDir] is declared final but is reassigned`

**Comment by henriquenfaria** [comment]
Did some testing on a project running Gradle version 4.10.2 and it worked fine.
Feel free to merge it.

**Comment by vanniktech** [comment]
Thanks for all of your work!
